File: src/java/tree/102.binary-tree-level-order-traversal.java
class P102 {
    // @lc app=leetcode id=102 lang=java
    // @lc code=start
    public class Solution {
        public java.util.List<java.util.List<Integer>> levelOrder(TreeNode root) {
            java.util.List<java.util.List<Integer>> out = new java.util.ArrayList<>();
            if (root==null) return out;
            java.util.ArrayDeque<TreeNode> q = new java.util.ArrayDeque<>();
            q.add(root);
            while(!q.isEmpty()){
                int sz = q.size();
                java.util.List<Integer> level = new java.util.ArrayList<>(sz);
                for(int i=0;i<sz;i++){
                    TreeNode n = q.poll();
                    level.add(n.val);
                    if (n.left!=null) q.add(n.left);
                    if (n.right!=null) q.add(n.right);
                }
                out.add(level);
            }
            return out;
        }
    }
    // @lc code=end
}

File: src/python/tree/102.binary-tree-level-order-traversal.py
from typing import List, Optional
from lc_test_utils import TreeNode

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []
        res = []
        q = [root]
        while q:
            level = []
            for _ in range(len(q)):
                node = q.pop(0)
                level.append(node.val)
                if node.left: q.append(node.left)
                if node.right: q.append(node.right)
            res.append(level)
        return res
